src/Data/Aeson/Types/ToJSON.hs

```diff
@@ -176,6 +176,9 @@ class GToJSON' enc arity f where
     -- and 'liftToEncoding' (if the @arity@ is 'One').
     gToJSON :: Options -> ToArgs enc arity a -> f a -> enc
 
+    gOmitField :: proxy enc -> proxy' arity -> f a -> Bool
```

Comment:
Is idea of having `gOmitField` and `gOmittedDefault` to inherit the default value in wrapper types like `newtype`s or singular `data Wrapper = Wrap SomeOtherType` wrappers?

I don't think it's necessary.

It also is (or at least was) to prefer single method `G*` classes so they are inlined better.

TL;DR I'd rather not add generics for `omitField` nad `omittedField`. It will make patch smaller and we won't need to worry about generically derived code performance.

Reply:
Okay, I'll see about removing them :-)

Reply:
Done.